Ruslan_Neshta/8/DungeonSaver.py

```diff
@@ -0,0 +1,126 @@
+import json
+import pickle
+from DungeonLogger import debugger_output
+from DungeonLogger import main_logger
+
+SAVE = 'DungeonGame'
+
+
+@debugger_output()
+def save_game(game, type='b'):
+    """
+    Saves game into a file
+
+    :param game: dict of in game objects
+    :param type: b for binary, j for json formats
+    :return: nothing
+    :rtype: None
+    """
+
+    if type == 'b':
+        save_pickle(game)
+
+    elif type == 'j':
+        save_json(game)
+
+    else:
+        raise RuntimeError('invalid type "{}" got'.format(type))
+
+
+@debugger_output()
+def save_pickle(game):
+    """
+    Saves game into binary format
+
+    :param game: dict of in game objects
+    :return: nothing
+    :rtype: None
+    """
+
+    with open(SAVE, 'wb') as save:
+        pickle.dump(game, save, protocol=3)
+
+
+@debugger_output()
+def save_json(game):
+    """
+    Saves game into json format
+
+    :param game: dict of in game objects
+    :return: nothing
+    :rtype: None
+    """
+
+    name = SAVE + '.json'
+
+    with open(name, 'w') as save:
+        json.dump(game, save)
+
+
+@debugger_output()
+def load_game():
+    """
+    Loads game from file
+
+    :return: dict of in game objects
+    :rtype: dict
+    """
+
+    try:
+        game = load_pickle()
+
+    except FileNotFoundError:
+        game = load_json()
+    
+    except FileNotFoundError:
```

Comment:
why do you need to handle FileNotFoundError two times?

Reply:
`load_json` can raise it